src/lib.rs

```diff
@@ -90,41 +96,107 @@ pub struct IronOxide {
+    pub rotations_needed: EitherOrBoth<UserId, Vec<GroupId>>,
+}
+
+impl PrivateKeyRotationCheckResult {
+    pub fn user_rotation_needed(&self) -> Option<UserId> {
+        match &self.rotations_needed {
+            EitherOrBoth::Left(u) | EitherOrBoth::Both(u, _) => Some(u.clone()),
+            _ => None,
+        }
+    }
+
+    //    pub fn group_rotation_needed() -> Option<Vec<GroupId>> {
+    //        unimplemented!()
+    //    }
+
+    //    pub fn rotate_all(ironoxide: &IronOxide) -> (UserId, Vec<GroupId>) //TODO consider when group private key rotation is added
+}
+
 /// Initialize the IronOxide SDK with a device. Verifies that the provided user/segment exists and the provided device
 /// keys are valid and exist for the provided account. If successful returns an instance of the IronOxide SDK
 pub fn initialize(device_context: &DeviceContext) -> Result<IronOxide> {
-    // 1 MB
-    const BYTES_BEFORE_RESEEDING: u64 = 1 * 1024 * 1024;
     let mut rt = Runtime::new().unwrap();
-    let account_id = device_context.account_id();
-    rt.block_on(crate::internal::user_api::user_key_list(
+    rt.block_on(crate::internal::user_api::user_get_current(
         &device_context.auth(),
-        &vec![account_id.clone()],
     ))
-    .and_then(|mut users| {
-        users
-            //We're using remove here because we don't actually need this HashMap anymore and remove
-            //returns us ownership so we can avoid a clone.
-            .remove(&device_context.account_id())
-            .map(|current_user_public_key| IronOxide {
-                recrypt: Recrypt::new(),
-                device: device_context.clone(),
-                user_master_pub_key: current_user_public_key,
-                rng: Mutex::new(ReseedingRng::new(
-                    rand_chacha::ChaChaCore::from_entropy(),
-                    BYTES_BEFORE_RESEEDING,
-                    EntropyRng::new(),
-                )),
-            })
-            .ok_or(IronOxideErr::InitializeError)
+    .and_then(|current_user| {
+        Ok(IronOxide::create_with_side_effects(
+            &current_user,
+            device_context,
+        ))
     })
+    .map_err(|_| IronOxideErr::InitializeError)
+}
+
+/// Initialize the IronOxide SDK and check to see if the user that owns this `DeviceContext` is
+/// marked for private key rotation, or if any of the groups that the user is an admin of is marked
+/// for private key rotation.
+pub fn initialize_check_rotation(device_context: &DeviceContext) -> Result<InitAndRotationCheck> {
+    Runtime::new().unwrap().block_on(
+        internal::user_api::user_get_current(device_context.auth()).and_then(|curr_user| {
+            let ironoxide = IronOxide::create_with_side_effects(&curr_user, &device_context);
+
+            if curr_user.needs_rotation() {
+                Ok(InitAndRotationCheck::RotationNeeded(
+                    ironoxide,
+                    PrivateKeyRotationCheckResult {
```

Comment:
What if any groups need rotation?

Reply:
That capability will not exist until https://github.com/IronCoreLabs/ironoxide/issues/45 is complete.